src/test/java/io/getunleash/repository/HttpFeatureFetcherTest.java

```diff
@@ -0,0 +1,278 @@
+package io.getunleash.repository;
+
+import static com.github.tomakehurst.wiremock.client.WireMock.*;
+import static com.github.tomakehurst.wiremock.core.WireMockConfiguration.wireMockConfig;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThat;
+import static org.assertj.core.api.AssertionsForClassTypes.assertThatThrownBy;
+
+import com.github.jenspiegsa.wiremockextension.ConfigureWireMock;
+import com.github.jenspiegsa.wiremockextension.InjectServer;
+import com.github.jenspiegsa.wiremockextension.WireMockExtension;
+import com.github.jenspiegsa.wiremockextension.WireMockSettings;
+import com.github.tomakehurst.wiremock.WireMockServer;
+import com.github.tomakehurst.wiremock.core.Options;
+import io.getunleash.FeatureToggle;
+import io.getunleash.util.UnleashConfig;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URISyntaxException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+
+@ExtendWith(WireMockExtension.class)
+@WireMockSettings(failOnUnmatchedRequests = false)
+public class HttpFeatureFetcherTest {
+
+    @ConfigureWireMock Options options = wireMockConfig().dynamicPort();
+    @InjectServer WireMockServer serverMock;
```

Comment:
Nice addition of wiremock. Been meaning to do this very thing for a while. Thanks!